tests/load/locustfiles/locustfile.py

```diff
@@ -131,19 +219,47 @@ def register(self) -> None:
             AssertionError: If the user fails to register a channel
             ValidationError: If the register message schema is not as expected
         """
+
         chid: str = str(uuid.uuid4())
 
         with self._time_event(name="register") as timer:
             body = json.dumps(dict(messageType="register", channelID=chid))
             self.ws.send(body)
             reply = self.ws.recv()
-            assert reply, f"No 'register' response CHID: {chid}"
             res: RegisterMessage = RegisterMessage(**json.loads(reply))
             assert res.status == 200, f"Unexpected status. Expected: 200 Actual: {res.status}"
             assert res.channelID == chid, f"Channel ID did not match, received {res.channelID}"
             timer.response_length = len(reply.encode("utf-8"))
         self.channels[chid] = res.pushEndpoint
 
-    @task
-    def do_nothing(self) -> None:
-        pass
+    @task(weight=95)
+    def send_notification(self):
+        """
+        Sends a notification to a registered enpoint
+
+        Raises:
+            AssertionError: If the server does not respond correctly (400, 500, etc)
+        """
+        stored = self.environment.parsed_options.notification_type == NotificationType.STORED
+        if stored:
+            self.disconnect()
+
+        channel_id = random.choice(list(self.channels.keys()))
+        endpoint_url = self.channels[channel_id]
+
+        self.disconnect()
+
+        with self.client.post(
+            url=endpoint_url,
+            name="Endpoint Notification",
+            data=self.encrypted_data,
+            headers=self.headers,
+        ) as response:
+            if response.status_code != 201:
+                response.failure(f"{response.status_code=}, expected 201, {response.text=}")
+                return
+
+        if stored:
+            self.connect()
+            self.hello()
+        self.ack()
```

Comment:
In the case where we send 'stored' messages, a failure to post would result in the user becoming disconnected because we return on line 260.

I solved this locally by extracting the post logic into another method, and checking for success before calling self.ack().

```suggestion
        notification_success: bool = self.post_notification(endpoint_url)

        if stored:
            self.connect()
            self.hello()

        if not notification_success:
            return

        self.ack()
```

```
    def post_notification(self, endpoint_url) -> bool:
        """
        Send a notification to Push

        Args:
            endpoint_url: A channel destination endpoint url
        Returns:
            bool: Flag indicating if the post notification request was successful
        """
        with self.client.post(
                url=endpoint_url,
                name="notification",
                data=self.encrypted_data,
                headers=self.headers,
        ) as response:
            if response.status_code != 201:
                response.failure(f"{response.status_code=}, expected 201, {response.text=}")
                return False
        return True
```

Reply:
Could we just ```else```  and not return? 
```suggestion
        with self.client.post(
            url=endpoint_url,
            name="Endpoint Notification",
            data=self.encrypted_data,
            headers=self.headers,
        ) as response:
            if response.status_code == 201:
                 if stored:
                     self.connect()
                     self.hello()
                 self.ack()
             else:   
                 response.failure(f"{response.status_code=}, expected 201, {response.text=}")
```

Reply:
I think for the else to work it would have to look like this right 😬?
```
if response.status_code == 201:
    if stored:
        self.connect()
        self.hello()
    self.ack()
else:
    response.failure(f"{response.status_code=}, expected 201, {response.text=}")
    if stored:
        self.connect()
        self.hello()
```

Reply:
Oh so we need to run ```connect``` and ```hello``` even on a failure? If the endpoint doesn't return a 201 do we need to reconnect or should we just hope the user (through locust) tries to send another notification and when that passes we reconnect and send the ```ack```

Reply:
Leaving the users connected is part of the load we are trying to re-create is my understanding and in future when we have more tasks, what if the next task chosen isn't 'send_notification', we would be leaving the user in an unexpected state.